kasa/tests/test_smartdevice.py

```diff
@@ -22,11 +22,13 @@
 ]
 
 
+@device_iot
```

Comment:
Thanks for the review @rytilahti. I know it was a lot and I thought hard about how to make it smaller but couldn’t come up with anything that made much sense. 

Most of the comments are agreeable and doable and I will hopefully get sorted Monday. There is one question I have to clarify with you first. I restricted these tests with `@device_iot` as a short term measure to keep the PR from getting bigger (yes I know :-). Currently too many tests access sysinfo and break. 

So I think we have two options:

1. Move all the SMART fixtures into a different directory so they don’t run on all tests.
2. Restrict them for now with `@device_iot` and then gradually re-enable test_file by test_file as we introduce the interfaces. 

I prefer 2. because it will mean we have a single test suite and a good interface. What do you think?

Reply:
Hmm, so those tests that test for specific payloads (like those schema checks in test_smartdevice) should only be tested against "device_iot"/"legacy" payloads, and indeed worth guarding for the time being.

Even then, we probably want to keep the "smart" fixtures separated from the legacy ones, so creating a new directory to host them from the get-go is a good idea if that's simple to achieve. If not, let's keep it simple for the time being and revisit it later.

But maybe the more generic ones (like testing for state_attributes structure, tests for cli outputs, ..) could still test everything if doing so doesn't require too many changes?